pkg/proxy/nftables/proxier.go

```diff
@@ -0,0 +1,1542 @@
+	masqueradeMark := fmt.Sprintf("%#08x", masqueradeValue)
+	klog.V(2).InfoS("Using iptables mark for masquerade", "ipFamily", ipFamily, "mark", masqueradeMark)
+
+	serviceHealthServer := healthcheck.NewServiceHealthServer(hostname, recorder, nodePortAddresses, healthzServer)
+
+	var nftablesFamily nftables.Family
+	if ipFamily == v1.IPv4Protocol {
+		nftablesFamily = nftables.IPv4Family
+	} else {
+		nftablesFamily = nftables.IPv6Family
+	}
+
+	proxier := &Proxier{
+		svcPortMap:          make(proxy.ServicePortMap),
+		serviceChanges:      proxy.NewServiceChangeTracker(newServiceInfo, ipFamily, recorder, nil),
+		endpointsMap:        make(proxy.EndpointsMap),
+		endpointsChanges:    proxy.NewEndpointChangeTracker(hostname, newEndpointInfo, ipFamily, recorder, nil),
+		syncPeriod:          syncPeriod,
+		nftablesFamily:      nftablesFamily,
+		nftables:            nftables.New(nftablesFamily, kubeProxyTable),
+		masqueradeAll:       masqueradeAll,
+		masqueradeMark:      masqueradeMark,
+		exec:                utilexec.New(),
+		localDetector:       localDetector,
+		hostname:            hostname,
+		nodeIP:              nodeIP,
+		recorder:            recorder,
+		serviceHealthServer: serviceHealthServer,
+		healthzServer:       healthzServer,
+		nodePortAddresses:   nodePortAddresses,
+		networkInterfacer:   proxyutil.RealNetwork{},
+		conntrackTCPLiberal: conntrackTCPLiberal,
+	}
+
+	burstSyncs := 2
+	klog.V(2).InfoS("Iptables sync params", "ipFamily", ipFamily, "minSyncPeriod", minSyncPeriod, "syncPeriod", syncPeriod, "burstSyncs", burstSyncs)
+	proxier.syncRunner = async.NewBoundedFrequencyRunner("sync-runner", proxier.syncProxyRules, minSyncPeriod, syncPeriod, burstSyncs)
+
+	return proxier, nil
+}
+
+// NewDualStackProxier creates a MetaProxier instance, with IPv4 and IPv6 proxies.
+func NewDualStackProxier(
+	sysctl utilsysctl.Interface,
+	syncPeriod time.Duration,
+	minSyncPeriod time.Duration,
+	masqueradeAll bool,
+	masqueradeBit int,
+	localDetectors [2]proxyutiliptables.LocalTrafficDetector,
+	hostname string,
+	nodeIPs map[v1.IPFamily]net.IP,
+	recorder events.EventRecorder,
+	healthzServer healthcheck.ProxierHealthUpdater,
+	nodePortAddresses []string,
+) (proxy.Provider, error) {
+	// Create an ipv4 instance of the single-stack proxier
+	ipv4Proxier, err := NewProxier(v1.IPv4Protocol, sysctl,
+		syncPeriod, minSyncPeriod, masqueradeAll, masqueradeBit, localDetectors[0], hostname,
+		nodeIPs[v1.IPv4Protocol], recorder, healthzServer, nodePortAddresses)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create ipv4 proxier: %v", err)
+	}
+
+	ipv6Proxier, err := NewProxier(v1.IPv6Protocol, sysctl,
+		syncPeriod, minSyncPeriod, masqueradeAll, masqueradeBit, localDetectors[1], hostname,
+		nodeIPs[v1.IPv6Protocol], recorder, healthzServer, nodePortAddresses)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create ipv6 proxier: %v", err)
+	}
+	return metaproxier.NewMetaProxier(ipv4Proxier, ipv6Proxier), nil
+}
+
```

Comment:
```suggestion
// Base chains are those that are registered into the 
// https://wiki.nftables.org/wiki-nftables/index.php/Configuring_chains#Adding_base_chains
```

Reply:
ah, yeah, I should probably document better...